nautobot/core/jobs/cleanup.py

```diff
@@ -77,7 +95,8 @@ def run(self, *, cleanup_types, max_age=None):
 
             if CleanupTypes.JOB_RESULT in cleanup_types:
                 self.logger.info("Deleting JobResult records prior to %s", cutoff)
-                _, deleted_dict = JobResult.objects.restrict(self.user, "delete").filter(date_done__lt=cutoff).delete()
+                queryset = JobResult.objects.restrict(self.user, "delete").filter(date_done__lt=cutoff)
```

Comment:
I still wonder if (in addition to the recursive approch you've added) we should *also* slice the initial queryset by time intervals as an additional boon to scalability? (E.g. if the cutoff is 30 days ago, and the oldest record in the queryset is 100 days ago, delete all records older than 90 days, then 80 days, then 70 days, etc. instead of trying to delete all of them at once?)

Reply:
Just using a `queryset.only("id").delete()` was enough to delete